### ISSUE TITLE
Exception in shape_changer.js

### ISSUE THREAD

**Issue by pmoore603** [SalieriC/SWADE-Immersive-Macros#53]
Actor with shape shift ability, fails when changing from one creature directly to another (i.e. no revert back to normal form in between).  To reproduce, actor starts in normal form, runs script to change into a creature, then run script again to change into another creature.

In this example, Sapa Kangee (actor) successfully changed into a Cat.  You can see "Cat (Sapa Kangee)" under [SWIM] Shape Changing actor folder.  Then, when changing directly into a Fox, it creates the Fox in the same folder, but then the exception occurs:  Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'bar1'):

<img width="1302" alt="Screen Shot 2022-03-31 at 1 34 08 PM" src="https://user-images.githubusercontent.com/62725574/161119729-dbbf1109-c8b5-4138-a84f-85c07c447aec.png">

I see what the problem is, and I will submit a fix.

**Comment by SalieriC** [comment]
Don't worry, I'm working on it, so that'll only result in merging conflicts. We can save that work. I think it's just because `carry` isn't defined, so just adding `let carry` will solve the problem. Or did you spot another issue?

**Comment by pmoore603** [comment]
You got it.
But, I also have some refactors in the code... should be able to remove quite a few lines.

**Comment by SalieriC** [comment]
That's awesome. But pull before you do so, otherwise you'll have to merge the conflicts. =)

**Comment by pmoore603** [comment]
Regarding `set_tokenSettings()`:  Turns out that in both cases (pcID or no pcID), the data values were coming from `actor`.
Regarding `update_preset()`:  I basically created a pointer for the second set of updates being saved, again based on pcID vs. no pcID.  I moved the `name` up to the first set of data, since it may as well be based on `pc`, and the first set is all `pc` related data.  Used the pointer also for the `arcaneBackground` set of updates.
